tests/func/test_dataset_query.py

```diff
@@ -459,6 +459,77 @@ def test_order_by_limit(cloud_test_catalog, save, animal_dataset):
     ]
 
 
+@pytest.mark.parametrize("save", [True, False])
+@pytest.mark.parametrize(
+    "cloud_type,version_aware",
+    [("s3", True)],
+    indirect=True,
+)
+def test_limit(cloud_test_catalog, save, animal_dataset):
```

Comment:
[Q] Any reason you've added these tests instead of testing the exposes DataChain API?

Reply:
I was thinking it is more close to the internals, since exposed DataChain API still uses the internal methods.